### ISSUE TITLE
Buttons of an imageless `wxTreeCtrl` are too small at high DPI

### ISSUE THREAD

**Issue by taler21** [wxWidgets/wxWidgets#24640]
At high DPI, the size of the buttons in a `wxTreeCtrl` is too small when the tree is used without images.

This issue can be reproduced with the following changes to the aui sample:

```patch
diff --git "a/samples/aui/auidemo.cpp" "b/samples/aui/auidemo.cpp"
index 56f5384bb6..c8b7dacc45 100644
--- "a/samples/aui/auidemo.cpp"
+++ "b/samples/aui/auidemo.cpp"
@@ -1631,7 +1631,7 @@ wxTreeCtrl* MyFrame::CreateTreeCtrl()
     wxVector<wxBitmapBundle> images;
     images.push_back(wxArtProvider::GetBitmapBundle(wxART_FOLDER, wxART_OTHER, size));
     images.push_back(wxArtProvider::GetBitmapBundle(wxART_NORMAL_FILE, wxART_OTHER, size));
-    tree->SetImages(images);
+    //tree->SetImages(images);
 
     wxTreeItemId root = tree->AddRoot("wxAUI Project", 0);
     wxArrayTreeItemIds items;

```

At a DPI scaling of 200%, the buttons size is correct with the unmodified sample (with images in the `wxTreeCtrl`).

![aui_dpi-200_tree-with-images](https://github.com/wxWidgets/wxWidgets/assets/99262969/e42d6a44-eecd-40b4-80c7-fed01fa6e18c)

But **without images** in the `wxTreeCtrl` the **buttons** are **too small**.

![aui_dpi-200_tree-without-images](https://github.com/wxWidgets/wxWidgets/assets/99262969/0d42d8de-c3f9-4045-b56a-819870a118de)

At a DPI scaling of 300%, the buttons are much too small in an imageless `wxTreeCtrl`.

![aui_dpi-300_tree-with-vs-without-images](https://github.com/wxWidgets/wxWidgets/assets/99262969/97064bba-17e7-432a-80c5-8445c84c4c10)

### Platform and version information

- wxWidgets version  you use: 3.2.5 and master (d1506f40c67453ad5ad6799dc6e71a7e71ff56f4)
- wxWidgets port  you use: wxMSW
- OS  and its version: Windows 10

**Comment by vadz** [comment]
I didn't really look at this, but I'd be surprised if we have any way to change the size of the expander, i.e. I'm afraid it's the native control which does it (incorrectly).

**Comment by dsa-t** [comment]
Looks like a `TVM_SETINDENT` message needs to be sent with a `FromDIP(16)` value at creation and at DPI changes.

See https://bugs.eclipse.org/bugs/show_bug.cgi?id=480586

**Comment by vadz** [comment]
Yes, indeed, this is confirmed in [the original bug report](https://developercommunity.visualstudio.com/t/explorer-themed-cmfc-ctreectrl-not-high-dpi-aware/46919) too.

We should be able to do this in `wxTreeCtrl::MSWUpdateFontOnDPIChange()` without changing ABI, i.e. even in 3.2.
